pkg/prometheus/resource_selector.go

```diff
@@ -152,20 +152,14 @@ func (rs *ResourceSelector) SelectServiceMonitors(ctx context.Context, listFn Li
 				break
 			}
 
-			for _, rl := range endpoint.RelabelConfigs {
-				if rl.Action != "" {
```

Comment:
I'm not sure why we had this test in the first place. I've modified `validateRelabelConfig()` to default to the replace action if the field is empty.